objects/src/assets/token_symbol.rs

```diff
@@ -0,0 +1,79 @@
+use super::{AssetError, Felt, StarkField, ToString};
+use crate::utils::string::String;
+
+const MAX_SUMBOL_LENGHT: usize = 6;
+#[derive(Clone, Copy, Debug)]
+pub struct TokenSymbol(Felt);
+
+impl TokenSymbol {
+    pub fn new(symbol: &str) -> Result<Self, AssetError> {
+        let felt = encode_symbol_to_felt(symbol)?;
+        Ok(Self(felt))
+    }
+
+    pub fn as_felt(&self) -> Felt {
+        self.0
+    }
```

Comment:
nit: I'd change this to `to_felt()`.

But we can also replace this by implementing `impl From<&TokenSymbol> for Felt`.